include/circt/Dialect/Moore/MIRExpressions.td

```diff
@@ -133,14 +133,15 @@ def ShrOp : ShiftOp<"shr"> {
 
 class EqualOp<string name> : MIROp<name, [
   Pure,
-  Commutative
+  Commutative,
+  SameTypeOperands
 ]> {
   let arguments = (ins SimpleBitVectorType:$lhs,
                        SimpleBitVectorType:$rhs,
                        UnitAttr:$mode);
   let results = (outs I1:$result);
   let assemblyFormat = [{
-    ( `case` $mode^ )? $lhs `,` $rhs attr-dict `:` type($lhs) `,` type($rhs)
+    ( `case` $mode^ )? $lhs `,` $rhs attr-dict `:` functional-type($lhs,$result)
   }];
 }
```

Comment:
I think I forgot to delete the `EqualOp` here. There's a replacement op in `MooreOps.td` (or similar), I think called `EqOp`. You should be able to just delete this.

Reply:
I know it and agree to delete it. But now the condition of the scf::if needs I1 as its argument. I attempted to use the `SingleBitType` you defined, but `the type mismatch with I1` will be thrown.